packages/expo-camera/android/src/main/java/expo/modules/camera/CameraViewModule.kt

```diff
@@ -139,11 +144,51 @@ class CameraViewModule : Module() {
     OnDestroy {
       try {
         moduleScope.cancel(ModuleDestroyedException())
-      } catch (e: IllegalStateException) {
+      } catch (_: IllegalStateException) {
         Log.e(TAG, "The scope does not have a job in it")
       }
     }
 
+    Class("Picture", PictureRef::class) {
+      Property("width") { picture: PictureRef ->
+        picture.ref.width
+      }
+
+      Property("height") { picture: PictureRef ->
+        picture.ref.height
+      }
+
+      AsyncFunction("savePictureAsync") { picture: PictureRef, options: SavePictureOptions?, promise: Promise ->
+        val image = picture.ref
+        val response = Bundle()
+
+        cacheDirectory.let {
+          response.apply {
+            putInt("width", image.width)
+            putInt("height", image.height)
+          }
+
+          val quality = options?.quality ?: 0
+          // Cache compressed image in imageStream
+          ByteArrayOutputStream().use { imageStream ->
+            image.compress(Bitmap.CompressFormat.JPEG, quality.toInt() * 100, imageStream)
+            // Write compressed image to file in cache directory
+            val filePath = writeStreamToFile(it, imageStream)
+            image.recycle()
+            val imageFile = File(filePath)
+            val fileUri = Uri.fromFile(imageFile).toString()
+            response.putString("uri", fileUri)
+
+            // Write base64-encoded image to the response if requested
+            if (options?.base64 == true) {
+              response.putString("base64", Base64.encodeToString(imageStream.toByteArray(), Base64.NO_WRAP))
```

Comment:
Do we need to save the image on disk when returning the `base64` encoding? People probably won't need the file if they request `base64` 🤔

Reply:
It's just what we've always done. We do it in `expo-image-manipulator` to https://github.com/expo/expo/blob/aad8591fcf2093da0f83df6f40c69eb972e6d8bd/packages/expo-image-manipulator/ios/ImageManipulatorModule.swift#L84. I guess we never had an option to return without writing to disk before